js/backup.js

```diff
@@ -0,0 +1,659 @@
+    var sanitized = sanitizeFileName(name);
+    return new Promise(function(resolve, reject) {
+      parent.getFile(sanitized, {create: true, exclusive: true}, function(file) {
+        return file.createWriter(function(writer) {
+          resolve(writer);
+        }, reject);
+      }, reject);
+    });
+  }
+
+  function readFileAsText(parent, name) {
+    return new Promise(function(resolve, reject) {
+      parent.getFile(name, {create: false, exclusive: true}, function(fileEntry) {
+        fileEntry.file(function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            resolve(e.target.result);
+          };
+          reader.onerror = reject;
+          reader.onabort = reject;
+          reader.readAsText(file);
+        }, reject);
+      }, reject);
+    });
+  }
+
+  function readFileAsArrayBuffer(parent, name) {
+    return new Promise(function(resolve, reject) {
+      parent.getFile(name, {create: false, exclusive: true}, function(fileEntry) {
+        fileEntry.file(function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            resolve(e.target.result);
+          };
+          reader.onerror = reject;
+          reader.onabort = reject;
+          reader.readAsArrayBuffer(file);
+        }, reject);
+      }, reject);
+    });
+  }
+
+  function getAttachmentFileName(attachment) {
+    return attachment.fileName || (attachment.id + '.' + attachment.contentType.split('/')[1]);
+  }
+
+  function readAttachment(parent, message, attachment) {
+    var name = getAttachmentFileName(attachment);
+    var sanitized = sanitizeFileName(name);
+    var attachmentDir = message.received_at;
+    return new Promise(function(resolve, reject) {
+      parent.getDirectory(attachmentDir, {create: false, exclusive: true}, function(dir) {
+        return readFileAsArrayBuffer(dir, sanitized ).then(function(contents) {
+          attachment.data = contents;
+          return resolve();
+        }, reject);
+      }, reject);
+    });
+  }
+
+  function writeAttachment(dir, attachment) {
+    var filename = getAttachmentFileName(attachment);
+    return createFileAndWriter(dir, filename).then(function(writer) {
+      var stream = createOutputStream(writer);
+      return stream.write(attachment.data);
+    });
+  }
+
+  function writeAttachments(parentDir, name, messageId, attachments) {
+    return createDirectory(parentDir, messageId).then(function(dir) {
+      return Promise.all(_.map(attachments, function(attachment) {
+        return writeAttachment(dir, attachment);
+      }));
+    }).catch(function(error) {
+      console.log(
+        'writeAttachments: error exporting conversation',
+        name,
+        ':',
+        error && error.stack ? error.stack : error
+      );
+      return Promise.reject(error);
+    });
+  }
+
+  function sanitizeFileName(filename) {
+    return filename.toString().replace(/[^a-z0-9.:+()'"#\- ]/gi, '_');
```

Comment:
That regular expression only leaves the listed characters - everything else is turned into an underscore.